src/Diagnostics.RuntimeHost/Services/DiagnosticTranslator/DiagnosticTranslatorService.cs

```diff
@@ -0,0 +1,288 @@
+                    for (int j = 0; j < allowedColumnIndexList.Count && allowedColumnIndexList[j] < columnCount; j++)
+                    {
+                        int columnIndex = allowedColumnIndexList[j];
+                        dataset.Table.Rows[i][columnIndex] = translatedText[i * allowedColumnCount + j];
+              
+                    }
+                }
+            }
+            return dataset;
+        }
+
+        public async Task<DiagnosticData> GetDataSummaryTranslation(DiagnosticData dataset, string language)
+        {
+            int nameColumnIndex = 0;
+            return await GetRenderingTranslation(dataset, language, new List<int> { nameColumnIndex });
+        }
+
+        public async Task<DiagnosticData> GetInsightsTranslation(DiagnosticData dataset, string language)
+        {
+            int messageColumnIndex = 1;
+            int insightNameColumnIndex = 2;
+            int insightValueColumnIndex = 3;
+            int insightSolutionColumnIndex = 5;
+
+            return await GetRenderingTranslation(dataset, language, new List<int> { messageColumnIndex, insightNameColumnIndex, insightValueColumnIndex, insightSolutionColumnIndex });
+        }
+
+        // Cognitive translation API has a limit of 100 elements for string arrays to be sent in request body. 
+        private async Task<List<string>> GetGroupTranslations(List<string> textsToTranslate, string languageToTranslate)
+        {
+            int translationGroupCount = textsToTranslate.Count / 100;
+            int lastGroupSize = textsToTranslate.Count % 100;
+            List<List<string>> translationGroupList = new List<List<string>>();
+            for (int i = 0; i <= translationGroupCount; i++)
+            {
+                int groupSize = i == translationGroupCount ? lastGroupSize : 100;
+                translationGroupList.Add(textsToTranslate.GetRange(i * 100, groupSize));
+            }
+
+            var translatedTextsGroup = await Task.WhenAll(translationGroupList.Select(textBatch => GetTranslations(textBatch, languageToTranslate)));
+            return translatedTextsGroup.Where(group => group != null).SelectMany(group => group).ToList();
+        }
+
+        // Translation method to make the http request with string arrays to be translated
+        public async Task<List<string>> GetTranslations(List<string> textsToTranslate, string languageToTranslate)
+        {
+            if (textsToTranslate == null)
+            {
+                return null;
+            }
+
+            // Always look for cache before calling translator API
+            Tuple<string, string> key = new Tuple<string, string>(languageToTranslate, string.Join("_", textsToTranslate));
+            if (this._translationCacheService.TryGetValue(key, out List<string> value))
+            {
+                return value;
+            }
+
+            string route = string.Format("/translate?api-version=3.0&from=en&to={0}", languageToTranslate);
```

Comment:
Could we add this route and querystring to appsettings? Similiar to what we do for ASC

```
"AzureSupportCenter": {
    "BaseUri": "https://api.diagnostics.msftcloudes.com",
    "ApiUri": "/api/diagnosis/",
    "ApiVersion": "2018-02-01",
    "UserAgent": "AppLensClient",
    "AADAuthority": "https://login.microsoftonline.com/microsoft.onmicrosoft.com",
    "TokenResource": "https://microsoft.onmicrosoft.com/azurediagnostic",
    "ClientId": "",
    "AppKey": "",
    "Enabled": true,
    "DiagAscHeader":  ""
  }
```

Reply:
Added query string and Api version into appsettings